src/ORM/Behavior/EnumBehavior.php

```diff
@@ -0,0 +1,150 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         5.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\ORM\Behavior;
+
+use ArrayObject;
+use BackedEnum;
+use Cake\Datasource\EntityInterface;
+use Cake\Event\EventInterface;
+use Cake\ORM\Behavior;
+use Cake\ORM\Entity;
+use Cake\ORM\Query;
+
+/**
+ * Enum behavior
+ *
+ * Allows you to map a specific field in your model to a backed enum
+ *
+ *  * Add behavior + config to connect a field with a backed enum
+ * ```
+ * $this->addBehavior('Enum', [
+ *     'fieldMap' => [
+ *         'published' => ArticleStatus::class
+ *     ]
+ * ]);
+ * ```
+ */
+class EnumBehavior extends Behavior
+{
+    /**
+     * Default config
+     *
+     * These are merged with user-provided configuration when the behavior is used.
+     *
+     * @var array<string, mixed>
+     */
+    protected array $_defaultConfig = [
+        'fieldMap' => [],
```

Comment:
Can we name this something clearer?

`fieldTypes` or `fieldEnums`?

A map is a super generic name.